services/housekeeper/housekeeper.go

```diff
@@ -248,6 +273,48 @@ func (hk *Housekeeper) UpdateProposerDutiesWithoutChecks(headSlot uint64) {
 	log.WithField("numDuties", len(_duties)).Infof("proposer duties updated: %s", strings.Join(_duties, ", "))
 }
 
+// When the update is triggered, we store registered validators in Redis for an the current and next epoch.
+// This allows us to maintain 1 epoch in advance, which is enough to check if a validator is registered for a given epoch.
+func (hk *Housekeeper) updateMevCommitValidatorRegistrations(headSlot uint64) {
+	// Only update once per epoch - at the epoch start of the epoch
+	if headSlot%common.SlotsPerEpoch == 0 {
+		return
+	}
+
+	epoch := headSlot / common.SlotsPerEpoch
+
+	duties, err := hk.beaconClient.GetProposerDuties(epoch)
```

Comment:
Isn't this already obtained in `hk.updateProposerDuties`?

Reply:
That stores a different representation.